docs/use-cases-provisioning.md

```diff
@@ -4,13 +4,40 @@ title: Infrastructure Provisioning
+## Motivation
 
-A lot of operations require some sort of locking to ensure that no more than one mutation is executed on a resource at a time.
-Temporal provides strong guarantees of uniqueness by business Id. This can be used to implement such locking behavior in a fault tolerant and scalable manner.
+Provisioning resources on a public or hybrid cloud is a potentially long-running operation with many possibilities for intermittent failures. While existing deployment tools support simple operations, many scenarios may still require a custom provisioning flow:
 
-Some real-world use cases:
+- Automatic infrastructure provisioning for a new customer in multi-tenant environments.
+- Particularly large deployments when tens or even hundreds of thousands of resources should be configured.
+- Provisioning of custom resources that are not supported by off-the-shelf tools.
+- Complex configuration logic that is determined at deployment time.
 
- * [Using Temporal workflows to spin up Kubernetes, by Banzai Cloud](https://banzaicloud.com/blog/introduction-to-cadence/)
- * [Using Temporal to orchestrate cluster life cycle in HashiCorp Consul, by HashiCorp](https://www.youtube.com/watch?v=kDlrM6sgk2k&feature=youtu.be&t=1188)
+It's beneficial to have a single workflow engine to manage all the various tasks: spinning up the cluster, long term monitoring, managing upgrades, database schema migrations, automated staged rollout of new features.
+
+Some provisioning operations may take dozens of minutes. Ad-hoc solutions may fail in the middle and leave the system in an undefined state. 
+
+## Benefits of Temporal
+
+Temporal workflows can express complex decision trees using a general-purpose programming language. Support of long-running operations, polling, responding to events, automatic retries are excellent building blocks for robust provisioning flow. If an extensive provisioning workflow fails in the middle, Temporal would handle the error and restart the flow at the right spot.
+
+Temporal can route activity execution to a specific process or host, which is useful for many provisioning scenarios.
+
+Many resource management operations require locking to ensure that only one mutation is executed on a resource at a time. Temporal provides strong guarantees of uniqueness by operation identifier. This primitive enables the implementation of locking behavior in a fault-tolerant and scalable manner.
+
+## Example Scenarios
+
+**CI/CD**. Implementing CI/CD pipelines and deploying applications to containers or virtual or physical machines is a non-trivial process. Its business logic has to deal with complex requirements around rolling upgrades, canary deployments, and rollbacks. Temporal is a perfect platform for building a deployment solution because it provides all the necessary guarantees and abstractions, allowing developers to focus on business logic.
```

Comment:
```suggestion
**CI/CD**. Implementing CI/CD pipelines and deploying applications to containers or virtual or physical machines is a non-trivial process. The logic has to deal with complex requirements around rolling upgrades, canary deployments, and rollbacks. Temporal is a perfect platform for building a deployment solution because it provides all the necessary guarantees and abstractions, allowing developers to focus on business logic.
```